docs/developer-docs/integrations/ledger/icp-index-local-setup.md

```diff
@@ -0,0 +1,233 @@
+```
+
+### Step 7: Deploy the ICP index canister:
+Here it is assumed that the canister id of your local ICP ledger is `ryjl3-tyaaa-aaaaa-aaaba-cai`, otherwise replace it with your ICP ledger canister id. 
+
+```
+dfx deploy icp_index --specified-id qhbym-qaaaa-aaaaa-aaafq-cai --argument '(record {ledger_id = principal "ryjl3-tyaaa-aaaaa-aaaba-cai"})'
+```
+
+The ICP index canister will start synching right away and will automatically try to fetch new blocks from the ICP ledger every few seconds. 
+
+### Step 8: Check the status and ICP ledger id on the ICP index canister.
+You can check that the correct ledger id was set but runnig the following command.
+```
+dfx canister call qhbym-qaaaa-aaaaa-aaafq-cai ledger_id '()'
+```
+The result is the ledger canister id that the index canister is using to sync.
+```
+(principal "ryjl3-tyaaa-aaaaa-aaaba-cai")
+```
+
+```
+dfx canister call qhbym-qaaaa-aaaaa-aaafq-cai status '()'
+```
+It should return something like this:
+```
+(record { num_blocks_synced = 1 : nat64 })
+```
+Depending on how many mint operations you created while setting up your ICP ledger the number of synched blocks here will be 0 if no initial balances were parsed or `X` if `X` initial balances were parsed. In the case of this tutorial the guide on setting up a local ledger was used and there only one initial balance was parsed as an initialization argument. Hence, the number of blocks synched at this stage is 1. 
+
+### Step 9: Create some new blocks to sync.
+
+You can check that the synchronization of the index is working by creating some transaction on the ICP ledger and then checking the status.
+If you followed the guide on setting up an ICP ledger locally your default identity should have some ICP to be send. 
+Send some ICP to any prinicpal.
+```
+dfx canister call ryjl3-tyaaa-aaaaa-aaaba-cai icrc1_transfer '(record { to = record { owner = principal "npki3-wdfh4-siaeq-orwh4-bh5of-r7mxr-i35lm-6f2eh-rtmwp-dmzmn-tae";};  amount = 100000:nat;})'
+```
+
+```
+dfx canister call qhbym-qaaaa-aaaaa-aaafq-cai status '()'
+```
+It should now indicate that an additional block was synced compared to the last time we called the status endpoint. You may have to wait a couple of seconds for the index canister to sync. 
+```
+(record { num_blocks_synced = 2 : nat64 })
+```
+
+### Step 10: Fetch some blocks
+You can use the ICP index canister to fetch blocks like so.
+You have to specify the block at which you want to start fetching from (i.e. the lowest index you want to fetch). If you want to start from the beginning you have to set start to 0. Similarly, the length parameter indicates the number of blocks you would like to fetch. Since the last status call indicated that there are two blocks that were synced you can set this to 2. Note that if you specify more than 2 blocks it will simply return the maximum number of blocks the index contains (The limit of blocks per call is usually set to 2000 blocks).
```

Comment:
```suggestion
You have to specify the block at which you want to start fetching from (i.e. the lowest index you want to fetch). If you want to start from the beginning you have to set `start` to 0. Similarly, the `length` parameter indicates the number of blocks you would like to fetch. Since the last status call indicated that there are two blocks that were synced, you can set this to 2. Note that if you specify more than 2 blocks it will simply return the maximum number of blocks the index contains (the limit of blocks per call is usually set to 2000 blocks).

```